engine/src/main/java/org/terasology/rendering/dag/ConditionDependentNode.java

```diff
@@ -53,7 +53,7 @@ private boolean checkConditions() {
     public void propertyChange(PropertyChangeEvent evt) {
         boolean conditionsChanged = checkConditions();
         if (conditionsChanged) {
-            refreshTaskList(); // TODO: Think about `pending` mode for the nodes, for not refreshing task list more than once
+            requestTaskListRefresh(); // TODO: Think about `pending` mode for the nodes, for not refreshing task list more than once
```

Comment:
You can remove the comment now. 

On the other hand, while AbstractNode no longer needs a WorldRenderer reference ConditionDependentNode now certainly does. But its subclasses won't, so you can keep some as they are in this PR.